internal/tools/aws/database.go

```diff
@@ -184,22 +185,35 @@ func (d *RDSDatabase) rdsDatabaseProvision(installationID string, logger log.Fie
 		return err
 	}
 
-	encryptionKey, err := d.client.kmsCreateSymmetricKey(awsID, "Key used for encrypting RDS database")
+	keyAliasName := KMSAliasNameRDS(awsID)
+	keyMetadata, err := d.client.kmsGetSymmetricKey(keyAliasName)
 	if err != nil {
-		return errors.Wrapf(err, "unable to create RDS encryption key for installation %s", installationID)
-	}
+		if !IsErrorCode(err, kms.ErrCodeNotFoundException) {
+			return errors.Wrapf(err, "unable to verify encryption key %s for the RDS cluster ID %s", keyAliasName, awsID)
+		}
 
-	err = d.client.kmsCreateAlias(*encryptionKey.KeyId, KMSAliasNameRDS(awsID))
-	if err != nil && !IsErrorCode(err, kms.ErrCodeAlreadyExistsException) {
-		deletionKeyErr := d.client.kmsScheduleKeyDeletion(*encryptionKey.KeyId, KMSMinTimeEncryptionKeyDeletion)
-		if deletionKeyErr != nil {
-			logger.WithError(deletionKeyErr).Errorf("Failed to schedule encryption key %s for deletition", *encryptionKey.KeyId)
+		keyMetadata, err = d.client.kmsCreateSymmetricKey("Key used for encrypting RDS database")
+		if err != nil {
+			return errors.Wrapf(err, "unable to create an encryption key for the RDS cluster ID %s", awsID)
 		}
 
-		return errors.Wrapf(err, "unable to create a RDS encryption key alias name for installation %s", installationID)
+		err = d.client.kmsCreateAlias(*keyMetadata.KeyId, keyAliasName)
+		if err != nil {
+			scheduleErr := d.client.kmsScheduleKeyDeletion(*keyMetadata.KeyId, KMSMinTimeEncryptionKeyDeletion)
+			if scheduleErr != nil {
+				logger.WithError(scheduleErr).Errorf("Failed to schedule deletion for the encryption key %s", keyAliasName)
+			}
+			return errors.Wrapf(err, "unable to create an encryption key alias name for the RDS cluster ID %s", awsID)
+		}
```

Comment:
This is the part that confuses me. If I understand the flow of this PR, it is:

1. Lookup KMS key alias
2. I that doesn't exist, create a new KMS key
3. Create an alias for that new key
4. If that fails, try to quickly schedule the key for deletion and long any errors.

The problem I see with this is that keys can still leak if step 2 works, but 3 fails.

This seems to stem from this design choice: https://github.com/mattermost/mattermost-cloud/blob/a424a80e2ac9d9545760bdc620cf0ecd60e86e76/internal/tools/aws/kms.go#L21-L22

If we were to create a tag on the KMS key when it was created, we should be able to look it up on follow-up attempts and not need to try to schedule it for deletion if anything goes wrong. This is how we make the other AWS flows resilient to failure. Is there a reason we can't do the same thing here?